types/react-native/test/index.tsx

```diff
@@ -652,6 +652,11 @@ export class FlatListTest extends React.Component<FlatListProps<number>, {}> {
   );
 
   render() {
+    const { ListEmptyComponent } = this.props;
+    const listEmptyComponent: JSX.Element | null | undefined = React.isValidElement(ListEmptyComponent)
+        ? ListEmptyComponent
+        : ListEmptyComponent && <ListEmptyComponent />;
```

Comment:
`ListEmptyComponent` is not narrowed to `ReactElement<unknown>` if a union member is `ReactElement<any>`. But with the change in this PR the full type is  `ReactElement<unknown> | ...` so `isValidElement` does narrow it again.